9.3.0 Release (#4304)
cypress-io/cypress-documentation: content/_changelogs/9.3.0.md

  [our blog post](https://cypress.io/blog/2022/01/18/uploading-files-with-selectfile/).
  Fixes [#19524](https://github.com/cypress-io/cypress/issues/19524) and
  [#170](https://github.com/cypress-io/cypress/issues/170).
- Exposed the `Buffer` polyfill, as [`Cypress.Buffer`](../api/utilities/buffer),
  to enable working with binary data. Addressed in
  [#19524](https://github.com/cypress-io/cypress/issues/19524).
- Added the `CYPRESS_DOWNLOAD_PATH_TEMPLATE` environment variable which can be
  set to a string template for building the download URL for the Cypress binary.
  This environment variable is useful for users who are downloading the Cypress
  binary from a proxy that is not one-to-one with the Cypress's default download
  url. More information can be found in the
  [Install Binary](../guides/getting-started/installing-cypress#Install-binary)
  documentation. Fixes
  [#15697](https://github.com/cypress-io/cypress/issues/15697).

**Bugfixes:**

- Fixed an issue where the `--spec` CLI parameter was not working properly when
  passing multiple glob patterns that are separated by commas. Fixes
  [#16102](https://github.com/cypress-io/cypress/issues/16102).

> 

Reviewer: https://github.com/cypress-io/cypress/pull/19560 was missing a release

```suggestion
- CYPRESS_VERIFY_TIMEOUT can also be set in package.json and .npmrc (depending on os)
```

Reviewer 2: updated: [`d33b75a` (#4304)](https://github.com/cypress-io/cypress-documentation/pull/4304/commits/d33b75a15dbf6992109b8bf4faf0425da9768eb6)

## 9.3.0

_Released 1/18/2022_

**Features:**

- Added a new Cypress command, called
  [`.selectFile()`](../api/commands/selectfile), to select a file or files in an

>   HTML5 input element or simulates dragging a file or files into the browser.

Reviewer 3: ```suggestion
  HTML5 input element or simulate dragging a file or files into the browser.
```

## 9.3.0

_Released 1/18/2022_

**News:**

- We've
  [written a blog post](https://cypress.io/blog/2022/01/18/uploading-files-with-selectfile/)
  about the new Cypress command, `cy.selectFile()` introduced in `9.3.0`.

> 

Reviewer 2: @jennifer-shehane Sounds good. I pulled this section from a different changelog

**Features:**

> - Added a new Cypress command, called `cy.selectFile()`, which provide the

Reviewer 2: ```suggestion
- Added a new Cypress command, called `cy.selectFile()`, which provides the
```

> - Added a new Cypress command, called `cy.selectFile()`, which provides the
>   ability to submit a file to file input elements. This command can also
>   simulate a user drag-dropping files over the document. The API is similar to
>   `cypress-file-upload`, and we have provided a migration guide for previous
>   users of that plugin. For full details, see
>   [our blog post](https://cypress.io/blog/2022/01/18/uploading-files-with-selectfile/)
>   and the
>   [cy.selectFile() documentation](https://docs.cypress.io/api/commands/selectfile).
>   Fixes [#19524](https://github.com/cypress-io/cypress/issues/19524) and
>   [#170](https://github.com/cypress-io/cypress/issues/170).

Author: Let's match the wording from the selectFile doc itself.

Commands that are chained (child commands) do not have `cy.` prefixed.

Link to the migration guide where we mentioned the migration guide. 

```suggestion
- You can now selects a file or files in an HTML5 input element or simulates dragging a file
or files into the browser using `.selectFile()`. The API is similar to
  `cypress-file-upload` and we have provided a migration guide for previous
  users of that plugin. For full details, see
  [our blog post](https://cypress.io/blog/2022/01/18/uploading-files-with-selectfile/)
  and the
  [.selectFile() documentation](https://docs.cypress.io/api/commands/selectfile).
  Fixes [#19524](https://github.com/cypress-io/cypress/issues/19524) and
  [#170](https://github.com/cypress-io/cypress/issues/170).
```

Author: Also these docs links need to be updated to internal links.

> - Exposed the `Buffer` polyfill, as `Cypress.Buffer`, to enable working with

Author: Link to the new Cypress.Buffer doc here.

>   binary data. Addresses in

Reviewer 2: ```suggestion
  binary data. Addressed in
```

  [#19524](https://github.com/cypress-io/cypress/issues/19524).

> - Added `CYPRESS_DOWNLOAD_PATH_TEMPLATE` environment variable which can be set

Reviewer 2: ```suggestion
- Added the `CYPRESS_DOWNLOAD_PATH_TEMPLATE` environment variable which can be set
```

  to a string template for building the download URL for the Cypress binary.

>   This environment variable is useful for users who are downloading from a proxy

Reviewer 2: ```suggestion
  This environment variable is useful for users who are downloading the Cypress binary from a proxy
```

>   that is not one-to-one with the default download path of the Cypress binary.

Reviewer 2: ```suggestion
  that is not one-to-one with the Cypress's default download url.
```

  url. More information can be found in the

>   [Install Binary](https://docs.cypress.io/guides/getting-started/installing-cypress#Install-binary).

Author: Use internal links, 

```suggestion
  [Install Binary](guides/getting-started/installing-cypress#Install-binary).
```

  Fixes [#15697](https://github.com/cypress-io/cypress/issues/15697).

**Bugfixes:**

> - Update the `--spec` CLI parameter to accept multiple glob patterns separated

Reviewer 2: ```suggestion
- Updated the `--spec` CLI parameter to accept multiple glob patterns that are separated
```

>   by a comma. Fixes

Reviewer 2: ```suggestion
  by commas. Fixes
```

> - Updated the `--spec` CLI parameter to accept multiple glob patterns that are
>   separated by commas. Fixes
>   [#16102](https://github.com/cypress-io/cypress/issues/16102).

Author: This sounds more like a feature, but it was a fix

```suggestion
- Fixed an issue where the `--spec` CLI parameter was not working properly when passing multiple glob patterns that are
  separated by commas. Fixes
  [#16102](https://github.com/cypress-io/cypress/issues/16102).
```
